app/assets/javascripts/plate-map/fabric-events.js

```diff
@@ -323,72 +323,102 @@ var plateLayOutWidget = plateLayOutWidget || {};
       // get well value differences for each well in wellsHash
       getDifferentWellsVals: function(wellsHash) {
         var wells = [];
-        for (var wellIdx in wellsHash){
-          wells.push(wellsHash[wellIdx]);
+        for (var wellId in wellsHash){
+          wells.push(wellsHash[wellId]);
         }
-        var commonWell = this._getCommonWell(wells);
         var differentWellsVals = {};
-        var allFieldVal = {};
-        for (var fieldIdx = 0; fieldIdx < this.fieldList.length; fieldIdx++) {
-          allFieldVal[this.fieldList[fieldIdx].id] = [];
-        }
+        if (wells.length > 1){
+          var commonWell = this._getCommonWell(wells);
+          var allFieldVal = {};
+          for (var fieldIdx in wellsHash[0]) {
+            allFieldVal[fieldIdx] = [];
+          }
 
-        for (var wellIdx in wells){
-          var diffWellVal = {};
-          var curWellData = wells[wellIdx];
-          for (var fieldId in curWellData) {
-            var commonVal =commonWell[fieldId];
-            var curVal = curWellData[fieldId];
-            var newVal;
-            if (Array.isArray(curVal)) {
-              // get uncommonVal
-              newVal = [];
-              for (var idx = 0; idx < curVal.length; idx ++){
-                var curMultiVal = curVal[idx];
-                if (!this.containsObject(curMultiVal, commonVal)) {
-                  newVal.push(curMultiVal);
-                  if (!this.containsObject(curMultiVal, allFieldVal[fieldId])) {
-                    allFieldVal[fieldId].push(curMultiVal);
+          for (var wellIdx in wells){
+            var diffWellVal = {};
+            var curWellData = wells[wellIdx];
+            for (var fieldId in curWellData) {
+              var commonVal = commonWell[fieldId];
+              var curVal = curWellData[fieldId];
+              var newVal = null;
+              if (Array.isArray(curVal)) {
+                // get uncommonVal
+                newVal = [];
+                if (curVal[0] && typeof(curVal[0] === "object")){
```

Comment:
Shouldn't each entry in curVal be checked, not just the first?

Reply:
The way it works here is that if the multiplex field is empty, then the multiplex field in this well is an empty array. If the multiplex field has value, then the field will be [{subfield:val, subfield2:val2}, {subfield:val}]. I think checking the first entry is sufficient to be identified as a multiplexfield.